schematics/transforms.py

```diff
@@ -56,11 +56,17 @@ def import_loop(cls, instance_or_dict, field_converter, context=None,
     """
     if not isinstance(instance_or_dict, (cls, dict)):
         raise ModelConversionError('Model conversion requires a model or dict')
-    if mapping is None:
-        mapping = {}
-    data = dict(context) if context is not None else {}
-    errors = {}
 
+    if not env:
+        env = ConfigObject()
+    elif not isinstance(env, ConfigObject):
+        env = ConfigObject(env)
+    env._setdefault('partial', partial)
```

Comment:
I think it makes more sense to have values passed as arguments overwrite those in `env`.

Having two ways to specify options is confusing. But I understand you don't want to break backward compatibility.